### ISSUE TITLE
[TGG] Class1 Bug

### ISSUE THREAD

**Issue by SvenPeldszus** [eMoflon/emoflon#522]
Sometimes a TGGRule is named "Class1" in the generated code instead of their real name.

An eap-file with this bug has been uploaded to the "issuefiles" folder (issue522__TGGRule_Class1_Bug.eap)

The error occurs in the project GravityTGG. This error is already present in the GravityTGG.ecore file at GravityTGG > Rules > Class1.

**Comment by anthonyanjorin** [comment]
this is true!  I've had this problem a few times but somehow can't really reproduce it on demand.  @SvenPeldszus Do you have a way of reproducing this in a few minimal steps from a well-defined starting point?

**Comment by Arikae** [comment]
*Push*, have seen this bug so many times and it's very annoying. 
I always had the feeling that this is likely to happen if you hit the Enter button after creating the plain rule object without entering another name than 'Class1'. 
After that point, it is always called like that....

I'll try to investigate that further after tuesday...

**Comment by SvenPeldszus** [comment]
I haven't noticed such a relation explicitly. However I had a few times the case that the properties window did not open automatically after creating a new rule. I assume this will result in a similar behaviour.

Additionally I have two further notices.

At first if you have this bug one time it never repeats. For all newly created rules the plug-in behaves as expected. However, creating manually a new rule with this name is possible and results in problems during code generation.

At second notice, I was able to get rid of this Class1 some times. This was most times after renaming elements at various points in enterprise architect. Unfortunately I wasn't able to detect a procedure which works always. Last time I got rid of this bug was after exporting the TGG project to a XMI file and reimporting it.
